domain/src/main/java/be/xplore/githubmetrics/domain/config/SchedulingConfiguration.java

```diff
@@ -0,0 +1,81 @@
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import org.springframework.scheduling.support.CronTrigger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+
+@ConditionalOnProperty(
+        value = "app.scheduling.enable", havingValue = "true", matchIfMissing = true
+)
+@Configuration
+@EnableScheduling
+@EnableCaching
+public class SchedulingConfiguration implements SchedulingConfigurer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowRunsRequestScheduler.class);
+    private final WorkflowRunsUseCase workflowRunsUseCase;
+    private final JobsUseCase jobsUseCase;
+    private final SchedulingConfig schedulingConfig;
+    private final Map<String, ScheduledFuture<?>> workflowRunsJob = new HashMap<>();
+
+    public SchedulingConfiguration(WorkflowRunsUseCase workflowRunsUseCase, JobsUseCase jobsUseCase, SchedulingConfig schedulingConfig) {
+        this.workflowRunsUseCase = workflowRunsUseCase;
+        this.jobsUseCase = jobsUseCase;
+        this.schedulingConfig = schedulingConfig;
+    }
+
+    @Override
+    public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+        ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+        threadPoolTaskScheduler.setPoolSize(5);
+        threadPoolTaskScheduler.setThreadNamePrefix("scheduler-thread");
+        threadPoolTaskScheduler.initialize();
+        workflowRunsJob(threadPoolTaskScheduler);
+        jobsJob(threadPoolTaskScheduler);
+        taskRegistrar.setTaskScheduler(threadPoolTaskScheduler);
+    }
+
+    private void workflowRunsJob(TaskScheduler taskScheduler) {
+        this.workflowRunsJob.put("WorkFlowRunsJob", taskScheduler.schedule(
+                workflowRunsUseCase::retrieveAndExportWorkflowRuns,
+                triggerContext -> {
+                    String cronExp = schedulingConfig.workflowRunsInterval();
+                    LOGGER.warn("WorkflowRunsJob cron: {}", cronExp);
+                    return new CronTrigger(cronExp).nextExecution(triggerContext);
+                }));
+    }
+
+    private void jobsJob(TaskScheduler taskScheduler) {
+        this.workflowRunsJob.put("JobsJob", taskScheduler.schedule(
+                jobsUseCase::retrieveAndExportJobs,
+                triggerContext -> {
+                    String cronExp = schedulingConfig.jobsInterval();
+                    LOGGER.warn("JobsJob cron: {}", cronExp);
+                    return new CronTrigger(cronExp).nextExecution(triggerContext);
+                }));
+    }
+
+    private void cancelScheduledTask(String key) {
+        ScheduledFuture<?> future = this.workflowRunsJob.get(key);
+        if (future != null) {
+            future.cancel(true);
+        }
+    }
+
```

Comment:
I get why you put these functions in a `@Configuration` class but I feel like its more then configuration no? These functions will be called in different usecases making them more then just configuration?

Reply:
fixed